### PR TITLE
Implementation of !schedule command

### PR DESCRIPTION
**Pull request by wlvstv**
### What was done
Implemented a new command to show Dan's most current schedule. The [endpoint](https://decapi.me/twitter/latest/dangheesling?search=*Show%20Schedule%20for%20Week*) that it is pulled from was supplied by MetallQ and gives the most current tweet that follows a specific format. I then use some regex to parse this response and format it into a discord embed.

### Why was this done
This was made in response to a chat about adding some functionality to the discord bot.

### Risks
The formatting using regex requires a pretty specific pattern to work, straying too far from this could cause some less than ideal display but we'll just have to learn as we go!

### Dependency changes

- Added imports for the [re](https://docs.python.org/3/library/re.html) module (provides regex support)
- Added imports for the [aiohttp](https://docs.aiohttp.org/en/stable/) module (was included in the existing requirements, provides http support)

As always, feedback is welcome, I am certainly not a python professional!

Mike (Decoy)

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -52,4 +54,16 @@ async def bruh(ctx):
                              '<:dangBRUH:661420374390603776>']))
 
 
+@bot.command()
```
Comment (abernauer) [inline bot.py:57]: Good code style left two blank lines between functions definitions.

File: bot.py [hunk 2]
```diff
@@ -52,4 +54,16 @@ async def bruh(ctx):
...
+async def schedule(ctx):
+    """Returns the most current schedule"""
```
Comment (abernauer) [inline bot.py:59]: Nice docustring!

File: bot.py [hunk 3]
```diff
@@ -52,4 +54,16 @@ async def bruh(ctx):
...
+    async with aiohttp.ClientSession() as session:
```
Comment (abernauer) [inline bot.py:60]: Great you used a context manager here and below. This should ensure resources are freed up when the blocks of code are done executing.

File: bot.py [hunk 4]
```diff
@@ -52,4 +54,16 @@ async def bruh(ctx):
...
+        async with session.get('https://decapi.me/twitter/latest/dangheesling?search=*Show%20Schedule%20for%20Week*') as resp:
+            respText = await resp.text()
+            dateFor = re.search('(?:\*Show Schedule for Week\* of)\s+(.\S*)', respText).group(1)
```
Comment (abernauer) [inline bot.py:63]: The code works though I ran it against the text and a regex checker and have my doubts. I can explain how I interpret your regex. Other than that concern, I am OK with approving the PR and merging the code as the command output is successful on the bot. Then readdress the concern if the function or command breaks further down the line.

**Comment by abernauer** [comment]
Hey Mike! Great work and initiative on implementing the schedule command. Just address my last comment and if you feel OK with merging it despite my concerns I will go ahead and do that. Oh also for future PRs can you create a new branch on your fork, for that feature thanks. Enjoying collaborating on this.

File: bot.py (hunk 4 above)
Comment (wlvstv) [reply bot.py:63]: Hey! So here is an explanation of the regex:
**dateFor**
This search is first looking for the group with the text "*Show Schedule for week* of) and not capturing this group. then looking for a space and capturing another group that is the text until the next space. This is so we can capture the date that this schedule is for. In theory, we could just capture the whole title as is but I wanted to pull the date out incase we needed it for something else.
**formatS**
This is using the sub function of the re library, this is to substitute matching groups with a replacement which I am using to break the lines of the schedule into newlines. The first part of the regex is looking for a sequence of non-digit characters (emoji + space + MON/TUE/WED etc) followed be a colon, a potential space, then 1 or 2 digits which is then followed by AM, there is an or with the repeated pattern for PM. In the substitution, I am adding a new line before these groups to be able to break them without cutting off any descriptions Dan is using (for example if there is a podcast that day)

Let me know if this makes sense or if more explanation is needed.

Thanks!

**Comment by abernauer** [comment]
OK concerns were addressed thanks for the contribution !

**Pull request merged**